src/sparse/impl/KokkosSparse_spmv_impl.hpp

```diff
@@ -1001,46 +1002,79 @@ struct SPMV_MV_OpenMP_Functor {
   KOKKOS_INLINE_FUNCTION
   void operator() (const team_member& dev) const
   {
-    for(ordinal_type loop = 0; loop < rows_per_team; ++loop) {
-      const ordinal_type iRow = static_cast<ordinal_type>(dev.league_rank()) * rows_per_team + loop;
+    for(ordinal_type loop = 0; loop < rows_per_thread; ++loop) {
+      const ordinal_type iRow = static_cast<ordinal_type>(dev.league_rank()*dev.team_size() + dev.team_rank())*rows_per_thread + loop;
       if (iRow >= m_A.numRows ()) {
 	return;
       }
-      const KokkosSparse::SparseRowViewConst<AMatrix> row = m_A.rowConst(iRow);
-      const ordinal_type row_length = static_cast<ordinal_type> (row.length);
       ordinal_type remainder = vecRemainder;
-      size_t strides[2];
-      m_x.stride(strides);
-      const ordinal_type m_x_stride = static_cast<ordinal_type>(strides[1]);
-      m_y.stride(strides);
-      const ordinal_type m_y_stride = static_cast<ordinal_type>(strides[1]);
 
       // Split multivector into chunks of 16 vectors
       for(ordinal_type vecOffset = 0; vecOffset < numVecs - 15; vecOffset += 16) {
-	strip_mine_large<16>(row, iRow, row_length, vecOffset, m_x_stride, m_y_stride);
+	strip_mine<16>(iRow, vecOffset);
       } // Loop over numVecs
 
-      if(remainder > 7) {
-	strip_mine_large<8>(row, iRow, row_length, numVecs - remainder, m_x_stride, m_y_stride);
-	remainder -= 8;
-      }
+      switch(remainder) {
```

Comment:
Do we really need to instantiate all 16 of these?  Is the larger build time worth the performance?